jdtls.ext/com.microsoft.jdtls.ext.core/src/com/microsoft/jdtls/ext/core/PackageCommand.java

```diff
@@ -174,36 +181,66 @@
      */
     private static List<PackageNode> getContainers(PackageParams query, IProgressMonitor pm) {
         IJavaProject javaProject = getJavaProject(query.getProjectUri());
-
         if (javaProject != null) {
             try {
                 IClasspathEntry[] references = javaProject.getRawClasspath();
-                return Arrays.stream(references).map(entry -> {
-                    try {
-                        entry = JavaCore.getResolvedClasspathEntry(entry);
-                        IClasspathContainer container = JavaCore.getClasspathContainer(entry.getPath(), javaProject);
-                        // HACK: There is an initialization issue for the first container.
-                        if (container == null) {
-                            container = JavaCore.getClasspathContainer(entry.getPath(), javaProject);
-                        }
-
-                        if (container != null) {
-                            return new ContainerNode(container.getDescription(), container.getPath().toPortableString(), NodeKind.CONTAINER,
-                                    entry.getEntryKind());
-                        }
-                    } catch (CoreException e) {
-                        // Ignore it
-                    }
-                    return null;
-                }).filter(containerNode -> containerNode != null).collect(Collectors.toList());
+                List<PackageNode> result = Arrays.stream(references).filter(entry -> entry.getEntryKind() != IClasspathEntry.CPE_LIBRARY)
+                        .map(entry -> getNodeFromClasspathEntry(entry, javaProject, NodeKind.CONTAINER)).filter(containerNode -> containerNode != null)
+                        .collect(Collectors.toList());
+
+                List<IClasspathEntry> referenceLibraries = Arrays.stream(references).filter(entry -> entry.getEntryKind() == IClasspathEntry.CPE_LIBRARY)
+                        .collect(Collectors.toList());
+                if (referenceLibraries.size() > 0) {
+                    result.add(new ContainerNode("ReferencedLibraries", REFERENCED_LIBRARIES_PATH, NodeKind.CONTAINER, IClasspathEntry.CPE_CONTAINER));
```

Comment:
ContainerNode("ReferencedLibraries", REFERENCED_LIBRARIES_PATH, NodeKind.CONTAINER, IClasspathEntry.CPE_CONTAINER) should be a constant